opm/autodiff/MultisegmentWell_impl.hpp

```diff
@@ -907,7 +903,7 @@ namespace Opm
         // For reservoir rate control, the distr in well control is used for the
         // rate conversion coefficients. For the injection well, only the distr of the injection
         // phase is not zero.
-        const double scale = scalingFactor(comp_idx);
+        const double scale = scalingFactor(ebosCompIdxToFlowCompIdx(comp_idx));
```

Comment:
I don't really understand this change?

Reply:
The scalingFactor still expects the legacy component index. (Due to interaction with well controls)